### PR TITLE
Update developer guide with user stories and NFRs

### LINKED ISSUE #6: Add user stories and NFR to Developer Guide

### PR DESCRIPTION
**Pull request by Graphcalibur**
Fixes #6

### CODE REVIEW

**Comment by ThePrevailingOne** [comment]
Overall LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -312,9 +353,12 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2. Should be able to hold up to 1000 entries without a noticeable sluggishness in performance for typical usage.
```
Comment (ThePrevailingOne) [inline docs/DeveloperGuide.md:357]: Just a nitpick, but maybe can change "sluggishness" to "lag"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -270,14 +270,55 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* * *`  | new user                                                   | see usage instructions                                       | refer to instructions when I forget how to use the App                                                             |
+| `* * *`  | user                                                       | add companies                                                |                                                                                                                    |
+| `* * *`  | user                                                       | add events related to a company                              | keep track of all of a company’s events                                                                            |
+| `* * *`  | user                                                       | add contact people from a company                            | easily look up their contact details                                                                               |
```
Comment (ThePrevailingOne) [inline docs/DeveloperGuide.md:278]: This implies that contact people already exists in the company right?

Comment (ThePrevailingOne) [reply docs/DeveloperGuide.md:278]: Should we have a "add contact people to a company" or is it already included in the "edit details of a company"

**Review by ThePrevailingOne — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Graphcalibur) [reply docs/DeveloperGuide.md:278]: I think it should be add contact people to a company, I'll change that real quick

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Graphcalibur) [reply docs/DeveloperGuide.md:357]: Sure

**Pull request merged**